Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/search/SearchDialog.kt

```diff
@@ -0,0 +1,116 @@
+package com.mentalmachines.droidcon_boston.views.search
+
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AutoCompleteTextView
+import android.widget.ImageView
+import androidx.fragment.app.DialogFragment
+import com.mentalmachines.droidcon_boston.R
+import com.mentalmachines.droidcon_boston.utils.visibleIf
+
+/**
+ * A full screen dialog that is used to provide searching functionality in the app.
+ *
+ * @property[queryListener] A callback that should be used to be notified of any queries made by
+ * this dialog.
+ */
+class SearchDialog : DialogFragment() {
+    private var backButton: ImageView? = null
+    private var clearButton: ImageView? = null
+    private var searchInput: AutoCompleteTextView? = null
+
+    var queryListener: ((String) -> Unit)? = null
+
+    private val currentQuery: String
+        get() = searchInput?.text?.toString().orEmpty()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setStyle(DialogFragment.STYLE_NORMAL, R.style.FullScreenDialogStyle)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.dialog_search, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        backButton = view.findViewById(R.id.action_back)
+        clearButton = view.findViewById(R.id.action_clear)
+        searchInput = view.findViewById(R.id.search_input)
+
+        setupViewListeners()
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        val width = ViewGroup.LayoutParams.MATCH_PARENT
+        val height = ViewGroup.LayoutParams.MATCH_PARENT
+        dialog?.window?.setLayout(width, height)
+    }
+
+    /**
+     * Sets all the button click and text listeners relevant to the views within our search dialog.
+     */
+    private fun setupViewListeners() {
+        backButton?.setOnClickListener {
+            dismiss()
+        }
+
+        clearButton?.setOnClickListener {
+            searchInput?.setText("")
+        }
+
+        searchInput?.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
```

Comment:
can do something like this to get rid of these unused methods: http://manabreak.eu/android/2016/09/17/quicktip-textwatcher-adapter.html
Surprised there's no built in adapter in Android APIs